components/compositing/touch.rs

```diff
@@ -81,13 +96,38 @@ impl TouchHandler {
 
         self.state = match self.state {
             Nothing => WaitingForScript,
-            Touching | Panning => Pinching,
+            Flinging { .. } => Touching,
+            Touching | Panning { .. } => Pinching,
             WaitingForScript => WaitingForScript,
             DefaultPrevented => DefaultPrevented,
             Pinching | MultiTouch => MultiTouch,
         };
     }
 
+    pub fn on_vsync(&mut self) -> Option<FlingAction> {
+        if let Flinging {
+            velocity,
+            ref cursor,
+        } = &mut self.state
+        {
+            if velocity.length().abs() >= FLING_MINIMUM {
+                // Todo: Probably we should multiply with the current refresh rate (and divide on each frame)
+                // or save a timestamp to account for a potentially changing display refresh rate.
+                *velocity *= FLING_SCALING_FACTOR;
+                debug_assert!(velocity.length() <= FLING_MAX && velocity.length() >= -FLING_MAX);
+                Some(FlingAction {
+                    delta: LayoutVector2D::new(velocity.x, velocity.y),
+                    cursor: *cursor,
+                })
+            } else {
+                self.state = Nothing;
+                None
```

Comment:
This could also be an early return  and then we save all the meaty logic for the main body of the function.